src/main/java/teammates/sqllogic/core/NotificationsLogic.java

```diff
@@ -0,0 +1,40 @@
+package teammates.sqllogic.core;
+
+import teammates.common.exception.EntityAlreadyExistsException;
+import teammates.common.exception.InvalidParametersException;
+import teammates.storage.sqlapi.NotificationsDb;
+import teammates.storage.sqlentity.Notification;
+
+/**
+ * Handles the logic related to notifications.
+ */
+public final class NotificationsLogic {
+
+    private static final NotificationsLogic instance = new NotificationsLogic();
+
+    private final NotificationsDb notificationsDb = NotificationsDb.inst();
```

Comment:
Pass in notificationsDb as a dependency instead (via initLogicDependencies). This way, it's easier to mock this out if we need to do unit testing (for the more complex logic methods, once we move the logic out of the db layer).